frontend/src/common/queries/wheresMyGene.ts

```diff
@@ -826,13 +826,18 @@ export function useMarkerGenes({
   }, [data]);
 
   return useQuery(
-    [USE_MARKER_GENES, cellTypeID, test],
+    /**
+     * (thuang): Add all arguments to `fetchMarkerGenes()` as dependencies,
+     * so React Query can cache responses correctly without running into
+     * issues like #4161
+     */
+    [USE_MARKER_GENES, cellTypeID, organismID, test, tissueID],
```

Comment:
@seve I added organismId to the cache key too, which I think makes sense to be part of the cache key. But please help verify that makes sense 🙏 Thanks!